app/src/main/java/io/homeassistant/companion/android/sensors/PhoneStateSensorManager.kt

```diff
@@ -41,16 +49,20 @@ class PhoneStateSensorManager : SensorManager {
 
     override val availableSensors: List<SensorManager.BasicSensor>
         get() = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP_MR1)
-            listOf(phoneState, sim_1, sim_2)
-        else listOf(phoneState)
+            listOf(phoneState, callNumber, sim_1, sim_2)
+        else listOf(phoneState, callNumber)
 
     override fun requiredPermissions(sensorId: String): Array<String> {
-        return arrayOf(Manifest.permission.READ_PHONE_STATE)
+        return if (sensorId == callNumber.id) {
+            arrayOf(Manifest.permission.READ_PHONE_STATE, Manifest.permission.READ_CALL_LOG)
+        } else arrayOf(Manifest.permission.READ_PHONE_STATE)
     }
 
     override fun requestSensorUpdate(
         context: Context
     ) {
+        /*We will update callNumber sensor together with phoneState sensor only when state is idle.
+        At active call callNumber sensor will be already updated by SensorReceiver.onReceive call intent*/
```

Comment:
Rather than do that why don't we get the phone state in this function then pass that down to `updatePhoneStateSensor` and a new function `resetCallNumber`?  That way we can decouple the two sensors.

Reply:
Done